### ISSUE TITLE
Incorrect result for SUM() and negative number 

### ISSUE THREAD

**Issue by mrigger** [cwida/duckdb#560]
Consider the following statements:
```sql
CREATE TABLE t0 (c0 INT);
INSERT INTO t0 VALUES (0);
SELECT SUM(-1) FROM t0; -- expected: {-1}, actual: {1.8446744073709552e+19}
```
Unexpectedly, the `SELECT` returns `1.8446744073709552e+19`. Since `t0` contains only a single row, I would expect `-1` to be fetched. I found this based on commit 3cf167887fa9cd375a4b1c67d0b427affa0a1430.

**Comment by Mytherin** [comment]
Thanks for the report! Fixed it in db0e6131397d4a09cd2c997fb86a9974b6c356e3. The problem was that there was an optimization for summing by constant that looked like this:

```cpp
sum += input * count;
```

But `count` was an unsigned number, thus `input * count` (which was `-1 * 1` in this case) was cast to an unsigned number, resulting in an underflow. Nice find!

**Comment by hsharaldsson** [comment]
I had a similar issue with some over partition by creating all kinds of wierd number in duckdb. Put my whole company off using duckdb. I would like to dig into my code and find an old commit and see if I can reproduce the error with a simple example. But it would be good to get some more information if similar problems have been fixed already like above might have been part of the issue. This ruined duckdb for me and the confidence in using it so fingers crossed it has been fixed.
